staging/src/k8s.io/kubectl/pkg/cmd/apply/apply.go

```diff
@@ -384,6 +384,11 @@ func (o *ApplyOptions) Validate() error {
 		return fmt.Errorf("--force cannot be used with --prune")
 	}
 
+	// Currently do not support pruning objects which are server-side applied.
+	if o.Prune && o.ServerSideApply {
+		return fmt.Errorf("objects created by server-side apply can not be pruned for historical reasons")
```

Comment:
I want the message to be about prune, not SSA.

e.g. "--prune is in beta and doesn't currently work on objects created by SSA"

Reply:
Updated--please have a look.

Reply:
ok, this works for me. You'll have to fix the test.